### PR TITLE
SPOPT-1569 - health status for Cassandra & Redis plugins

### PR DESCRIPTION
**Pull request by mpundlik**

### CODE REVIEW

File: db/sql/cassandra/plugin_impl_cassa.go [hunk 1]
```diff
@@ -15,11 +15,20 @@
...
+	"github.com/gocql/gocql"
+	"github.com/ligato/cn-infra/core"
 	"github.com/ligato/cn-infra/db/sql"
 	"github.com/ligato/cn-infra/flavors/local"
+	"github.com/ligato/cn-infra/health/statuscheck"
 	"github.com/willfaught/gockle"
 )
 
+// HealthCheck is a structure used to represent table to help verify health status
+type HealthCheck struct {
```
Comment (jozef-slezak) [inline db/sql/cassandra/plugin_impl_cassa.go]: Just do some read-only query using broker.Exec(). It is too much to have this not needed table on a particular schema.

File: db/sql/cassandra/plugin_impl_cassa.go [hunk 2]
```diff
@@ -73,18 +82,28 @@ func (p *Plugin) AfterInit() error {
 		p.session = gockle.NewSession(session)
 	}
 
-	/* TODO Register for providing status reports (polling mode)
+	// Register for providing status reports (polling mode)
 	if p.StatusCheck != nil && p.session != nil {
 		p.StatusCheck.Register(core.PluginName(p.String()), func() (statuscheck.PluginState, error) {
-			_, _, err := p.Skeleton.NewBroker("/").GetValue(healthCheckProbeKey, nil)
+			broker := p.NewBroker()
```
Comment (jozef-slezak) [inline db/sql/cassandra/plugin_impl_cassa.go]: just do broker.Exec("readonly cassandra query that is runalbe in every schema without having extra objects")

Comment (mpundlik) [reply db/sql/cassandra/plugin_impl_cassa.go]: done

File: examples/redis_plugin/main.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+// Redis flag to load config
+func init() {
+	flag.String("redis-config", "redis.conf",
+		"Location of the redis configuration file")
+}
+
+// ExampleFlavor is a set of plugins required for the redis example.
+type ExampleFlavor struct {
```
Comment (jozef-slezak) [inline examples/redis_plugin/main.go]: could you please move the example flavor & Deps struct to dedicated file inside example dir (Milan is currently doing that for all other examles)? thanks

File: db/keyval/redis/plugin_impl_redis.go [hunk 4]
```diff
@@ -57,6 +64,28 @@ func (p *Plugin) Init() error {
...
+// AfterInit is called by the Agent Core after all plugins have been initialized.
+func (p *Plugin) AfterInit() error {
+	if p.disabled {
+		return nil
+	}
+
+	// Register for providing status reports (polling mode)
+	if p.StatusCheck != nil {
+		p.StatusCheck.Register(core.PluginName(p.String()), func() (statuscheck.PluginState, error) {
+			_, _, err := p.Skeleton.NewBroker("/").GetValue(healthCheckProbeKey, nil)
```
Comment (jozef-slezak) [inline db/keyval/redis/plugin_impl_redis.go]: If possbile try to use directly connection broker to not create new instance again

File: db/sql/cassandra/plugin_impl_cassa.go (hunk 2 above)
Comment (jozef-slezak) [reply db/sql/cassandra/plugin_impl_cassa.go]: If possbile try to use directly connection broker to not create new instance again

Comment (mpundlik) [reply db/sql/cassandra/plugin_impl_cassa.go]: done

File: examples/redis_plugin/main.go (hunk 3 above)
Comment (mpundlik) [reply examples/redis_plugin/main.go]: done

**Pull request merged**